src/mmrelay/db_utils.py

```
@@ -771,24 +826,140 @@ def _store(cursor: sqlite3.Cursor) -> None:
+
+    if not stale_ids:
+        return 0
+
+    # Delete stale IDs in batches to avoid SQLite parameter limits
+    total_deleted = 0
+    chunk_size = 900  # Safe chunk size below SQLite's limit
+    for i in range(0, len(stale_ids), chunk_size):
+        chunk = stale_ids[i : i + chunk_size]
+        placeholders = ",".join("?" * len(chunk))
+        cursor.execute(
+            f"DELETE FROM {table} WHERE meshtastic_id IN ({placeholders})",
```

Ruff (0.15.4)

[error] Possible SQL injection vector through string-based query construction

(S608)